src/Microsoft.AspNetCore.Authentication.Core/AuthenticationSchemeProvider.cs

```diff
@@ -170,14 +130,6 @@ public virtual void AddScheme(AuthenticationScheme scheme)
                 {
                     _requestHandlers.Add(scheme);
                 }
-                if (typeof(IAuthenticationSignInHandler).IsAssignableFrom(scheme.HandlerType))
```

Comment:
> Its consistent with IAuthenticationRequestHandler, all AuthenticationHandlers implement Authenticate/Challenge/Forbid. We have one handler that implements SignIn, and two that implement SignOut.

The real difference is that `IAuthenticationRequestHandler` is still used and can be justified by performance reasons (`AuthenticationMiddleware.Invoke` avoids instantiating new handlers that don't implement `IAuthenticationRequestHandler`).

`IAuthenticationSignInHandler` and `IAuthenticationSignOutHandler` are now dead code with no use. I can't find any technical reason to keep them. It makes logging a PITA for implementers, it requires a cast for every request and offers 0 benefit.